File: app/src/main/webapp/js/classes/boolean-field.js
class BooleanField {
    static fromJson(json) {
        if (json == null) {
            return null;
        }

        const booleanField = new BooleanField();

        booleanField.setId(json.id);
        booleanField.setBitPosition(json.bitPosition);
        booleanField.setTrueDescription(json.trueDescription);
        booleanField.setFalseDescription(json.falseDescription);

        return booleanField;
    }

    static toJson(booleanField) {
        return {
            id:                 booleanField.getId(),
            bitPosition:        booleanField.getBitPosition(),
            trueDescription:    booleanField.getTrueDescription(),
            falseDescription:   booleanField.getFalseDescription()
        };
    }

    constructor() {
        this._id                = null;
        this._bitPosition       = null;
        this._trueDescription   = null;
        this._falseDescription  = null;
        this._fieldIndex        = null;
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setBitPosition(bitPosition) {
        this._bitPosition = bitPosition;
    }

    getBitPosition() {
        return this._bitPosition;
    }

    setTrueDescription(trueDescription) {
        this._trueDescription = trueDescription;
    }

    getTrueDescription() {
        return this._trueDescription;
    }

    setFalseDescription(falseDescription) {
        this._falseDescription = falseDescription;
    }

    getFalseDescription() {
        return this._falseDescription;
    }

    setFieldIndex(fieldIndex) {
        this._fieldIndex = fieldIndex;
    }

    getFieldIndex() {
        return this._fieldIndex;
    }
}

File: app/src/main/java/com/softwareverde/tidyduck/most/BooleanField.java
package com.softwareverde.tidyduck.most;

public class BooleanField {
    private Long _id;
    private String _bitPosition;
    private String _trueDescription;
    private String _falseDescription;

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getBitPosition() {
        return _bitPosition;
    }

    public void setBitPosition(String bitPosition) {
        _bitPosition = bitPosition;
    }

    public String getTrueDescription() {
        return _trueDescription;
    }

    public void setTrueDescription(String trueDescription) {
        _trueDescription = trueDescription;
    }

    public String getFalseDescription() {
        return _falseDescription;
    }

    public void setFalseDescription(String falseDescription) {
        _falseDescription = falseDescription;
    }
}
